src/mgr/MgrClient.cc

```diff
@@ -286,11 +280,6 @@ int MgrClient::start_command(const vector<string>& cmd, const bufferlist& inbl,
 
   ldout(cct, 20) << "cmd: " << cmd << dendl;
 
-  if (!session) {
-    lderr(cct) << "no session, waiting" << dendl;
-    wait_on_list(waiting_for_session);
-  }
```

Comment:
we don't throw them out; we queue them and they get sent later when teh connection happens.  the problem si that start_command() used to block, which made it uncallable from the handle_message context.